### PR TITLE
[enet] Update to 1.3.15

### LINKED ISSUE #11701: enet update to 1.3.15
Library name: enet

New version number: 1.3.15

https://github.com/lsalzman/enet/blob/master/ChangeLog

The last release dates back by two years. An update would be appreciated. There are NO breaking changes.

### PR DESCRIPTION
**Pull request by Fohlen**
**Describe the pull request**

- What does your PR fix? Fixes #11701 

- Which triplets are supported/not supported? Have you updated the CI baseline?
(No major changes to previous versions)

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

### CODE REVIEW

File: ports/enet/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: enet
-Version: 1.3.13-1
+Version: 1.3.15
 Homepage: https://github.com/lsalzman/enet
 Description: Reliable UDP networking library
```
Comment (NancyLi1013) [inline ports/enet/CONTROL:5]:
```suggestion
```
Could you please remove the blank line?

File: ports/enet/portfile.cmake [hunk 2]
```diff
@@ -1,9 +1,9 @@
 include(vcpkg_common_functions)
```
Comment (NancyLi1013) [inline ports/enet/portfile.cmake:1]:
```suggestion
```
This function is deprecated. Could you please remove this line?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/enet/portfile.cmake [hunk 3]
```diff
@@ -1,9 +1,13 @@
-include(vcpkg_common_functions)
+include(vcpkg_from_github)
+include(vcpkg_configure_cmake)
+include(vcpkg_install_cmake)
+include(vcpkg_copy_pdbs)
```
Comment (NancyLi1013) [inline ports/enet/portfile.cmake:4]: These functions are not needed here. Why do you add them?

Comment (Fohlen) [reply ports/enet/portfile.cmake:4]: I wasn't aware that these are automatically included (are they?). If that is the case ofc they can be removed.

Comment (Fohlen) [reply ports/enet/portfile.cmake]: @NancyLi1013 I removed these lines. Does it fit now? 
If so, best is probably to squash these commits.

Comment (NancyLi1013) [reply ports/enet/portfile.cmake]: Yes, just remove them.

File: ports/enet/portfile.cmake (hunk 2 above)
Comment (Fohlen) [reply ports/enet/portfile.cmake]: Done

**Comment by Fohlen** [comment]
Hey @NancyLi1013 
The tests seem to fail for obtrusive reasons. How would we go about fixing and merging this?

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
The failure has been resolved now.
LGTM, thanks for this PR @Fohlen .

**Pull request merged**